src/daft-core/src/join.rs

```diff
@@ -57,6 +57,7 @@ impl FromStr for JoinType {
     fn from_str(join_type: &str) -> DaftResult<Self> {
         match join_type {
             "inner" => Ok(Self::Inner),
+            "cross" => Ok(Self::Inner), // cross join is just inner join with no join keys
```

Comment:
We should still do the correct conversation but raise the error when created.

Reply:
Cross is not actually a join type but a join strategy for an inner join. We expose it as a strategy on the dataframe side but logically we do not represent it as a type. This match is essentially the boundary between the dataframe join types and the internal join types